include/dlaf/matrix/distribution.h

```diff
@@ -109,48 +113,48 @@ tile_element_index       O <-- I
 */
 
 class Distribution {
+  friend Distribution internal::create_single_tile_per_block_distribution(const Distribution& dist);
+
 public:
   /// Constructs a distribution for a non distributed matrix of size {0, 0} and block size {1, 1}.
   Distribution() noexcept;
 
-  /// Constructs a distribution for a non distributed matrix of size @p size and block size @p block_size.
-  ///
-  /// @param[in] element_offset is the element-wise offset of the top left tile of the matrix ,
-  /// @pre size.isValid(),
-  /// @pre !block_size.isEmpty().
-  Distribution(const LocalElementSize& size, const TileElementSize& block_size,
+  DLAF_DISTRIBUTION_DEPRECATED("Use another constructor")
```

Comment:
Worth writing a more informative deprecation message?

Reply:
This constructor shouldn't be deprecated... fixed it.